documentation/experimental_features.md

```diff
@@ -173,7 +174,62 @@ plan myplan(TargetSpec $targets) {
 ```
 
 In the example above, `target3` has to wait for `hostname` to finish on `target1` and `target2`
-before it can run `whoami`. The experimental `parallelize()` function accepts an array and a block,
+before it can run `whoami`.
+
+While useful, this form of parallelism is limited. Bolt plans have a few methods that allow a block
+of plan steps to execute in the background while other parts of the plan execute in parallel:
+* The `background` plan function begins executing a block of code in parellel with the main plan
+  and other backgrounded code blocks. This is great for use cases where you want to start a process
+  and don't care about the results, or don't need the results until much later in the plan. This
+  function returns a [Future](bolt_types_reference.md#Future) object so that the result can be
+  accessed later in the plan.
+* The `wait` function is a sister to `background`, it accepts a Future or array of Futures and
+  blocks until they are finished, optionally with a timeout, then returns the results.
+* The `parallelize` function accepts an array and a block of plan steps, and then creates a Future
+  for each block in the array. It blocks until all the Futures have finished.
+
+These two function invocations are equivalent:
+```
+# 
+parallelize(['./file1', './file2']) |$file| {
+  file_upload($file, '/home/user/', $targets)
+  ...
+}
+
+# Is equivalent to
+['./file1', './file2'].map |$file| {
+  background() || {
+    file_upload($file,...)
+    ...
+  }
+}
+```
```

Comment:
Shouldn't the second example have a `wait` call to make these functionally equivalent?

```
['./file1', './file2'].map |$file| {
  background() || {
    file_upload($file,...)
    ...
  }
}.wait
```

Reply:
Oh my gosh good eye! Definitely.